stream-tests/src/test/scala/org/apache/pekko/stream/scaladsl/SourceSpec.scala

```diff
@@ -311,19 +311,60 @@ class SourceSpec extends StreamSpec with DefaultTimeout {
     }
 
     "use decider when iterator throws" in {
+
```

Comment:
I'd prefer to have separate tests with descriptive titles if they test different edge cases, or at least a comment per run hinting what each is testing.